data/transactions/logic/assembler_test.go

```diff
@@ -2168,23 +2140,27 @@ func TestHasStatefulOps(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	t.Parallel()
 
-	source := "int 1"
-	ops := testProg(t, source, AssemblerMaxVersion)
-	has, err := HasStatefulOps(ops.Program)
-	require.NoError(t, err)
-	require.False(t, has)
+	for v := uint64(2); v <= AssemblerMaxVersion; v++ {
+		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
+			source := "int 1"
+			ops := testProg(t, source, v)
+			has, err := HasStatefulOps(ops.Program)
+			require.NoError(t, err)
+			require.False(t, has)
 
-	source = `txn Sender; int 1; app_opted_in; err`
-	ops = testProg(t, source, AssemblerMaxVersion)
-	has, err = HasStatefulOps(ops.Program)
-	require.NoError(t, err)
-	require.True(t, has)
+			source = `int 0; int 1; app_opted_in; err`
```

Comment:
question: changing to `int 0` to reference sender is to check cross version assembling for `proto("i:i")`, right?  For `proto("a:i")` is introduced after `directRefEnabledVersion`?

Reply:
Correct, it's convenient to use the mechanism that works for all versions, so we don't have to make multiple programs.